ledger-tool/src/bigtable.rs

```diff
@@ -336,6 +338,250 @@ pub async fn transaction_history(
+    debug!("from_slot: {}, to_slot: {}", from_slot, to_slot);
+
+    let source_bigtable = {
+        if args.source_is_emulator {
+            match solana_storage_bigtable::LedgerStorage::new_for_emulator(
+                &args.source_instance_name,
+                &args.source_app_profile_id,
+                &args.source_endpoint.unwrap(),
+                None,
+            ) {
+                Ok(bigtable) => bigtable,
+                Err(err) => {
+                    error!("Failed to connect to source bigtable: {:?}", err);
+                    return Err(Box::new(err));
+                }
+            }
+        } else {
+            match solana_storage_bigtable::LedgerStorage::new_with_config(
+                solana_storage_bigtable::LedgerStorageConfig {
+                    read_only: true,
+                    timeout: None,
+                    credential_type: CredentialType::Filepath(Some(
+                        args.source_credential_path.unwrap(),
+                    )),
+                    instance_name: args.source_instance_name,
+                    app_profile_id: args.source_app_profile_id,
+                },
+            )
+            .await
+            {
+                Ok(bigtable) => bigtable,
+                Err(err) => {
+                    error!("Failed to connect to source bigtable: {:?}", err);
+                    return Err(Box::new(err));
+                }
+            }
+        }
+    };
+
+    let destination_bigtable = {
+        if args.destination_is_emulator {
+            match solana_storage_bigtable::LedgerStorage::new_for_emulator(
+                &args.destination_instance_name,
+                &args.destination_app_profile_id,
+                &args.destination_endpoint.unwrap(),
+                None,
+            ) {
+                Ok(bigtable) => bigtable,
+                Err(err) => {
+                    error!("Failed to connect to destination bigtable: {:?}", err);
+                    return Err(Box::new(err));
+                }
+            }
+        } else {
+            match solana_storage_bigtable::LedgerStorage::new_with_config(
+                solana_storage_bigtable::LedgerStorageConfig {
+                    read_only: true,
+                    timeout: None,
+                    credential_type: CredentialType::Filepath(Some(
+                        args.destination_credential_path.unwrap(),
+                    )),
+                    instance_name: args.destination_instance_name,
+                    app_profile_id: args.destination_app_profile_id,
+                },
+            )
+            .await
+            {
+                Ok(bigtable) => bigtable,
+                Err(err) => {
+                    error!("Failed to connect to destination bigtable: {:?}", err);
+                    return Err(Box::new(err));
+                }
+            }
+        }
+    };
+
+    let (s, r) = unbounded::<u64>();
+    for i in from_slot..to_slot {
```

Comment:
I don't think this will fail very nicely if to_slot < from_slot.

Reply:
> I don't think this will fail very nicely if to_slot < from_slot.

correct! the `Range` will be constructed, but the iterator won't yield anything. i'd suggest testing during arg extraction

Reply:
fixed https://github.com/solana-labs/solana/pull/28122/commits/c8e8ccf0540f24f82be014f307aa50cd45355ec3